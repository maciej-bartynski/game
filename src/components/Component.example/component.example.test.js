import React, { useState as useStateMock } from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import Test from './component.example';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}))

describe('<Test />', () => {

  let mockedState = false;

  beforeEach(() => {
    const setState = jest.fn(() => mockedState = "state changed");
    useStateMock.mockImplementation(init => [init, setState]);
  });

  it('matches snapshot', () => {
    const tree = renderer.create(<Test />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('has 1 child', () => {
    const tree = renderer.create(<Test />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it('is pressed correctly', () => {
    const ComponentRendered = render(<Test />);
    const button = ComponentRendered.getByTestId('componentExample__button');
    expect(mockedState).not.toBe("state changed");
    fireEvent.press(button);
    expect(mockedState).toBe("state changed");
  });
});