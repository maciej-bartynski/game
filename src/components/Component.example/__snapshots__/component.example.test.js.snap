// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test /> matches snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderRadius": 5,
      "borderWidth": 1,
      "opacity": 1,
      "padding": 10,
    }
  }
  testID="componentExample__button"
>
  <Text>
     : 
    0
  </Text>
</View>
`;
